druid-shell/src/platform/gtk/window.rs

```diff
@@ -130,6 +129,8 @@ pub(crate) struct WindowState {
     /// Used to determine whether to honor close requests from the system: we inhibit them unless
     /// this is true, and this gets set to true when our client requests a close.
     closing: Cell<bool>,
+    /// A counter for generating unique tokens.
+    next_token: Cell<usize>,
```

Comment:
if this is an auto-incrementing counter, I would just bake it into the token type, and do something like `let token = TokenType::next()`. We even have a helper for this, `shell::common_util::Counter`.